engine/runtime/src/main/java/org/enso/interpreter/node/expression/foreign/CoerceNothing.java

```diff
@@ -22,8 +30,29 @@ public static CoerceNothing build() {
   public abstract Object execute(Object value);
 
   @Specialization(guards = "interop.isNull(value)")
-  public Object doNothing(Object value, @CachedLibrary(limit = "1") InteropLibrary interop) {
-    return EnsoContext.get(this).getBuiltins().nothing();
+  public Object doNothing(
+      Object value,
+      @CachedLibrary(limit = "1") InteropLibrary interop,
+      @CachedLibrary(limit = "3") WarningsLibrary warnings) {
+    var nothing = EnsoContext.get(this).getBuiltins().nothing();
+    return coerceNothing(value, nothing, warnings, nullWarningProfile);
+  }
+
+  public static Object coerceNothing(
```

Comment:
Nice solution of the code duplication problem.

Reply:
I dare to disagree. Why not put `CoerceNothing` node as `@Cached` value into `HostValueToEnsoNode.doNull` specialization? Besides, this uses a field, which is bad practice in Truffle. At least, we are unable to use `@GenerateUncached` on this node because of this field, that could easily be just a specialization parameter.

Reply:
The node wasn't used anywhere with `GenerateUncached`, so I thought a field is fine. I can change this to be a parameter if you think that is better.

I guess we can also just put this nested node.

Reply:
I suggest to get rid of this method, put everything into `doNothing` specialization, and definitely get rid of `nullWarningProfile` field, and rather put it as a parameter `@Cached("createCountingProfile()") ConditionProfile nullWarningProfile` to the specialization. And in `HostValueToEnsoNode`, just use `@Cached CoerceNothing` as parameter, and delegate to its `execute` method.

Reply:
Ok, thanks for the advice, I'll do